default/python/AI/ShipDesignAI.py

```diff
@@ -897,7 +902,7 @@ def update_stats(self, ignore_species=False):
                 else:
                     self.design_stats.stealth = 0
             elif partclass in FIGHTER_BAY:
-                self.design_stats.fighter_launch_rate += capacity
+                self.design_stats.fighter_launch_bays += 1
```

Comment:
Should probably keep a list/counter of bays instead and calculate the launch rate after all parts are processed. That would allow handling of future different bay parts, see comment below.

Reply:
Not sure how calculating the launch rate at a different moment helps with handling content changes.

Reply:
You want to query and use the ```part.capacity``` in case someone adds another bay part (say, with capacity / launch rate 3). That way, trivial content (i.e. stuff without an effects group) is automatically handled by the AI.

As the launch rate of at least 1 type of bay depends on the hangar type (i.e. the changes in your PR), we can not calculate the launch rate before we processed all other parts (and found out which hangar types there are in this design). So we need to defer that calculation. Keeping track of the number of launch bays is insufficient if there are different types of bays, so we want a list / counter of bays.